mars/dataframe/expressions/arithmetic/add.py

```diff
@@ -49,12 +51,22 @@ def level(self):
     def fill_value(self):
         return self._fill_value
 
+    @property
+    def scalar_value(self):
+        return self._scalar_value
+
 
 def add(df, other, axis='columns', level=None, fill_value=None):
-    op = DataFrameAdd(axis=axis, level=level, fill_value=fill_value)
+    if np.isscalar(other):
+        op = DataFrameAdd(axis=axis, level=level, fill_value=fill_value, scalar_value=other)
```

Comment:
Like I said, we don't need this branch actually.

Reply:
Instead, we need to check the type of `other`, make sure it's a `DataFrameType` or a scalar.